lib/msf/base/serializer/readable_text.rb

```diff
@@ -668,34 +659,41 @@ def self.dump_jobs(framework, verbose = false, indent = DefaultIndent, col = Def
       'Columns' => columns
       )
 
-    # jobs are stored as a hash with the keys being a numeric job_id.
-    framework.jobs.keys.sort{|a,b| a.to_i <=> b.to_i }.each { |k|
+    # jobs are stored as a hash with the keys being a numeric String job_id.
+    framework.jobs.keys.sort_by(&:to_i).each do |job_id|
       # Job context is stored as an Array with the 0th element being
       # the running module. If that module is an exploit, ctx will also
       # contain its payload.
-      ctx = framework.jobs[k].ctx
-      row = [ k, framework.jobs[k].name ]
-      row << (ctx[1].nil? ? (ctx[0].datastore['PAYLOAD'] || "") : ctx[1].refname)
-
-      # Make the LPORT show the bind port if it's different
-      local_port = ctx[0].datastore['LPORT']
-      bind_port = ctx[0].datastore['ReverseListenerBindPort']
```

Comment:
I doubt this was intentional, so no "tug of war" is going on :) The main question here will be: how do we want it represented on-screen?